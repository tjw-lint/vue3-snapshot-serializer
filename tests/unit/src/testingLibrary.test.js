import { userEvent } from '@testing-library/user-event';
import { render } from '@testing-library/vue';

import TestingLibrary from '@@/mockComponents/TestingLibrary.vue';

describe('TestingLibrary', () => {
  describe('Wrapper', () => {
    test('Renders default contents', () => {
      const wrapper = render(TestingLibrary);

      expect(wrapper)
        .toMatchSnapshot();
    });

    test('Contents change after toggle button clicked', async () => {
      const wrapper = render(TestingLibrary);

      const user = userEvent.setup();
      const button = wrapper.container.querySelector('[data-test="toggle"]');
      await user.click(button);

      expect(wrapper)
        .toMatchSnapshot();
    });
  });

  /**
   * On Containers, TLV always wraps the component in an extra div. Nothing we
   * can do about that:
   * https://testing-library.com/docs/vue-testing-library/api#container-htmlelement
   */

  describe('Container', () => {
    test('Renders default contents', () => {
      const { container } = render(TestingLibrary);

      expect(container)
        .toMatchSnapshot();
    });

    test('Contents change after toggle button clicked', async () => {
      const { container } = render(TestingLibrary);

      const user = userEvent.setup();
      const button = container.querySelector('[data-test="toggle"]');
      await user.click(button);
      const checkbox = container.querySelector('[data-test="checkbox"]');

      expect(container)
        .toMatchSnapshot('container');

      expect(checkbox)
        .toMatchSnapshot('checkbox');
    });
  });
});
