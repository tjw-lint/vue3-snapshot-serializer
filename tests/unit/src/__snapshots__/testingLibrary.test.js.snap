// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TestingLibrary > Container > Contents change after toggle button clicked > checkbox 1`] = `
<input
  checked="false"
  type="checkbox"
  value="'on'"
/>
`;

exports[`TestingLibrary > Container > Contents change after toggle button clicked > container 1`] = `
<div>
  <div>
    <button>
      Toggle availability
    </button>
    <input value="'false'" />
    <input
      checked="false"
      type="checkbox"
      value="'on'"
    />
    <h1 title="{ available: false }">
      Unavailable
    </h1>
  </div>
</div>
`;

exports[`TestingLibrary > Container > Renders default contents 1`] = `
<div>
  <div>
    <button>
      Toggle availability
    </button>
    <input value="'true'" />
    <input
      checked="true"
      type="checkbox"
      value="'on'"
    />
    <h1 title="{ available: true }">
      Available
    </h1>
  </div>
</div>
`;

exports[`TestingLibrary > Wrapper > Contents change after toggle button clicked 1`] = `
<div>
  <button>
    Toggle availability
  </button>
  <input value="'false'" />
  <input
    checked="false"
    type="checkbox"
    value="'on'"
  />
  <h1 title="{ available: false }">
    Unavailable
  </h1>
</div>
`;

exports[`TestingLibrary > Wrapper > Renders default contents 1`] = `
<div>
  <button>
    Toggle availability
  </button>
  <input value="'true'" />
  <input
    checked="true"
    type="checkbox"
    value="'on'"
  />
  <h1 title="{ available: true }">
    Available
  </h1>
</div>
`;
